elements/lisk-cryptography/src/keys.ts

```diff
@@ -81,3 +85,93 @@ export const getAddressFromPrivateKey = (privateKey: string): string => {
 
 	return getAddressFromPublicKey(publicKey);
 };
+
+const GENERATOR = [0x3b6a57b2, 0x26508e6d, 0x1ea119fa, 0x3d4233dd, 0x2a1462b3];
+
+const polymod = (uint5Array: number[]): number => {
+	let chk = 1;
+	for (const value of uint5Array) {
+		// eslint-disable-next-line no-bitwise
+		const top = chk >> 25;
+		// eslint-disable-next-line no-bitwise
+		chk = ((chk & 0x1ffffff) << 5) ^ value;
+		for (let i = 0; i < 5; i += 1) {
+			// eslint-disable-next-line no-bitwise
+			if ((top >> i) & 1) {
+				// eslint-disable-next-line no-bitwise
+				chk ^= GENERATOR[i];
+			}
+		}
+	}
+
+	return chk;
+};
+
+export const getBinaryAddressFromPublicKey = (publicKey: string): Buffer => {
+	const publicKeyBuffer = Buffer.from(publicKey, 'hex');
+	return hash(publicKeyBuffer).slice(0, 20);
+};
+
+export const createChecksum = (uint5Array: number[]): number[] => {
+	const values = uint5Array.concat([0, 0, 0, 0, 0, 0]);
+	// eslint-disable-next-line no-bitwise
+	const mod = polymod(values) ^ 1;
+	const result = [];
+	for (let p = 0; p < 6; p += 1) {
+		// eslint-disable-next-line no-bitwise
+		result.push((mod >> (5 * (5 - p))) & 31);
+	}
+	return result;
+};
+
+export const verifyChecksum = (integerSequence: number[]): boolean =>
+	polymod(integerSequence) === 1;
+
+export const getBase32AddressFromPublicKey = (
+	publicKey: string,
+	prefix: string,
+): string => {
+	const binaryAddress = getBinaryAddressFromPublicKey(publicKey);
+	const byteSequence = [];
+	for (const b of binaryAddress) {
+		byteSequence.push(b);
+	}
+	const uint5Address = convertUIntArray(byteSequence, 8, 5);
+	const uint5Checksum = createChecksum(uint5Address);
+
+	return `${prefix}${convertUInt5ToBase32(uint5Address.concat(uint5Checksum))}`;
+};
+
+const BASE32_ADDRESS_LENGTH = 41;
+const BASE32_CHARSET = 'zxvcpmbn3465o978uyrtkqew2adsjhfg';
+
+export const validateBase32Address = (address: string): boolean => {
+	if (address.length !== BASE32_ADDRESS_LENGTH) {
+		throw new Error(
+			'Address length does not match requirements. Expected 41 characters.',
+		);
+	}
+
+	const prefix = address.substring(0, 3);
+
+	if (prefix !== 'lsk') {
```

Comment:
default can be `lsk`, but to use it in other cases, prefix can be passed from the argument as done in above function.
`export const validateBase32Address = (address: string, prefix = 'lsk'): boolean`